fe/fe-core/src/main/java/org/apache/doris/common/MarkedCountDownLatch.java

```diff
@@ -28,11 +28,13 @@
 public class MarkedCountDownLatch<K, V> extends CountDownLatch {
 
     private Multimap<K, V> marks;
+    private Multimap<K, V> failedMarks;
```

Comment:
marks and failedMarks's had overlap effect  where they maybe both respent a failure;

use  com.google.common.collect.Table<K, V, Options<Boolean>> marks ?

for its Options<Boolean> value:
1.  equals Options.empty()  =>  no response;
2. Options.of(true) => reponse and succ;
3. Options.of(false) => response and failed;

then for function:
1. markedCountDown => Options.empty() => Options.of(true)
2. markedCountDownStatus => Options.empty() =>  Options.of(status.ok ? true : of)
3. getLeftMarks => those Options.empty() || Options.of(false);

Reply:
Currently, there is no specific requirement to differentiate between failed and unfinished cases, and countDownToZero may not be able to distinguish between them fully. Perhaps it would be worth considering your proposed solution if the need arises in the future.

Reply:
This PR needs to be picked to earlier versions, so keeping the changes as simple as possible will facilitate code review during the pick process. I think the optimizations can be addressed in a subsequent PR.